src/libraries/System.Runtime.InteropServices.JavaScript/gen/JsImportGenerator/JSExportCodeGenerator.cs

```diff
@@ -0,0 +1,253 @@
+                {
+                    marshallingNotSupportedCallback(p, e);
+                    return new EmptyJSGenerator();
+                }
+            }
+        }
+
+        public BlockSyntax GenerateJSExportBody()
+        {
+            StatementSyntax invoke = InvokeSyntax();
+            JSGeneratedStatements statements = JSGeneratedStatements.Create(_marshallers, _context, invoke);
+            bool shouldInitializeVariables = !statements.GuaranteedUnmarshal.IsEmpty || !statements.Cleanup.IsEmpty;
+            VariableDeclarations declarations = VariableDeclarations.GenerateDeclarationsForManagedToNative(_marshallers, _context, shouldInitializeVariables);
+
+            var setupStatements = new List<StatementSyntax>();
+            SetupSyntax(setupStatements);
+
+            if (!statements.GuaranteedUnmarshal.IsEmpty)
+            {
+                setupStatements.Add(MarshallerHelpers.Declare(PredefinedType(Token(SyntaxKind.BoolKeyword)), InvokeSucceededIdentifier, initializeToDefault: true));
+            }
+
+            setupStatements.AddRange(declarations.Initializations);
+            setupStatements.AddRange(declarations.Variables);
+            setupStatements.AddRange(statements.Setup);
+
+            var tryStatements = new List<StatementSyntax>();
+            tryStatements.AddRange(statements.Unmarshal);
+
+            tryStatements.AddRange(statements.InvokeStatements);
+
+            if (!statements.GuaranteedUnmarshal.IsEmpty)
+            {
+                tryStatements.Add(ExpressionStatement(AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                    IdentifierName(InvokeSucceededIdentifier),
+                    LiteralExpression(SyntaxKind.TrueLiteralExpression))));
+            }
+
+            tryStatements.AddRange(statements.NotifyForSuccessfulInvoke);
+            tryStatements.AddRange(statements.Marshal);
+
+            List<StatementSyntax> allStatements = setupStatements;
+            List<StatementSyntax> finallyStatements = new List<StatementSyntax>();
+            if (!statements.GuaranteedUnmarshal.IsEmpty)
+            {
+                finallyStatements.Add(IfStatement(IdentifierName(InvokeSucceededIdentifier), Block(statements.GuaranteedUnmarshal)));
+            }
+
+            finallyStatements.AddRange(statements.Cleanup);
+            if (finallyStatements.Count > 0)
+            {
+                allStatements.Add(
+                    TryStatement(Block(tryStatements), default, FinallyClause(Block(finallyStatements))));
+            }
+            else
+            {
+                allStatements.AddRange(tryStatements);
+            }
+
+            return Block(allStatements);
+        }
+
+        public BlockSyntax GenerateJSExportRegistration()
+        {
+            var registrationStatements = new List<StatementSyntax>();
+            // if (!RuntimeInformation.IsOSPlatform("browser")) return;
```

Comment:
nit: Do we want to leave these comments? They easily get out of sync with what is generated below.